examples/shadow/main.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/at-wat/mqtt-go"
+	"github.com/seqsense/aws-iot-device-sdk-go/v4"
+	"github.com/seqsense/aws-iot-device-sdk-go/v4/shadow"
+)
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	if len(os.Args) != 2 {
+		println("usage: shadow aws_iot_endpoint")
+		os.Exit(1)
+	}
+	host := os.Args[1]
+
+	for _, file := range []string{
+		"root-CA.crt",
+		"certificate.pem.crt",
+		"private.pem.key",
+	} {
+		_, err := os.Stat(file)
+		if os.IsNotExist(err) {
+			println(file, "not found")
+			os.Exit(1)
+		}
+	}
+
+	cli, err := awsiotdev.New(
+		"sample",
```

Comment:
Because the user needs to replace thingName, I think it's better to declare thing name as const and place it near the top of the file. (with the instruction that the user should replace it with their own one.)

Reply:
Changed to pass the thing name as a second argument of the example.